src/apps/reservation-list/modal/delete-reservation/delete-reservation-modal.tsx

```diff
@@ -7,65 +7,64 @@ import {
   useDeleteReservations,
   getGetReservationsV2QueryKey
 } from "../../../../core/fbs/fbs";
-import { useDeleteV1UserReservationsIdentifier } from "../../../../core/publizon/publizon";
-import { ReservationType } from "../../../../core/utils/types/reservation-type";
-import { isDigital } from "../../../loan-list/utils/helpers";
+import {
+  getGetV1UserReservationsQueryKey,
+  useDeleteV1UserReservationsIdentifier
+} from "../../../../core/publizon/publizon";
+import { isFaust, isIdentifier } from "../../../dashboard/util/helpers";
 
 interface DeleteReservationModalProps {
   modalId: string;
-  reservation: ReservationType;
+  reservations: string[];
 }
 
 const DeleteReservationModal: FC<DeleteReservationModalProps> = ({
   modalId,
-  reservation
+  reservations
 }) => {
   const t = useText();
   const queryClient = useQueryClient();
   const { mutate: deletePhysicalReservation } = useDeleteReservations();
   const { mutate: deleteDigitalReservation } =
     useDeleteV1UserReservationsIdentifier();
   const { close } = useModalButtonHandler();
-  const physicalDeletion = useCallback(() => {
-    if (!isDigital(reservation) && reservation.reservationId) {
+
+  const removeSelectedReservations = () => {
+    if (reservations.length > 0) {
+      const reservationsToDelete = reservations
+        .map((id) => Number(isFaust(id)))
+        .filter((id) => id !== 0);
+      const digitalMaterialsToDelete = reservations
+        .map((id) => isIdentifier(id))
+        .filter((id) => id !== null);
```

Comment:
Please consider whether map and filter are both needed.

If the reservation id is a Faust then we know it is a number. If it is an identifier then we know it is not null.

Reply:
The faustid is normally handled as a string in the system, but the api endpoint for deletion only accepts integers. That is the reason why we cast it as we do here, altho it might seem redundant. I removed the null check, but it's probably there for a reason.